src/pathtools.h

```diff
@@ -77,9 +77,38 @@ inline std::string make_path_relative_to_current_dir(const std::string& path
   auto p = fs::path{path};
   if (p.is_absolute() || p == "")
   {
-    return p;
+    return p.make_preferred().string();
+  }
+  return fs::relative(fs::absolute(filename).parent_path() / p).make_preferred().string();
+}
+
+/** Cleanup @p path, e.g. by adjusting and removing duplicate separators. 
+ * @param path A path given
+ * @return @p path in a canonical and native format.
+ **/
+inline std::string normalize_path(const std::string& path) {
+    fs::path p_canonical = fs::weakly_canonical(path);
+    std::string p_norm = p_canonical.make_preferred().string();
+    return p_norm;
+}
+
+/** Tries to find a home path, otherwise returns the root path.
+ * @return home path
+ **/
+inline fs::path get_home_dir()
+{
+  fs::path p_home = fs::current_path().root_path();
```

Comment:
I'm not sure about this fallback case ... why would we return `root_path()`?
This seems wrong to me.

I think we could take inspiration from [root_path()](https://en.cppreference.com/w/cpp/filesystem/path/root_path) and return `path()` as a fallback.

This way, the caller could check if there is a HOME directory or not.

We would of course also have to check this when we are using `get_home_dir()`.

Reply:
`root_path` is indeed not the best solution. How about `current_path()`? That's also easy to test and perhaps a better fallback. Say a file can also be located in the installation folder, and could then be found as a fallback

Reply:
> That's also easy to test

I disagree. It seems dangerous, because the current path might change at any point.

> and perhaps a better fallback.

I don't think the current path should be involved, it has nothing to do with the HOME directory.

If no HOME directory is defined, this function should basically return an error. If `path()` is returned, this would mean it didn't work and there is no HOME dir.

Reply:
Ok, I see! I updated to return `path()` now, in case where it fails